pkg/ebpf/tracer.go

```diff
@@ -656,6 +658,52 @@ func (t *Tracer) determineConnectionDirection(conn *ConnectionStats) ConnectionD
 	return OUTGOING
 }
 
+func (t *Tracer) determineConnectionIntraHost(connections []ConnectionStats) {
+
+	type connKey struct {
+		Address util.Address
+		Port    uint16
+		Type    ConnectionType
+	}
+
+	newConnKey := func(connStat *ConnectionStats, useRAddrAsKey bool) connKey {
+		key := connKey{Type: connStat.Type}
+		if useRAddrAsKey {
+			if connStat.IPTranslation == nil {
+				key.Address = connStat.Dest
+				key.Port = connStat.DPort
+			} else {
+				key.Address = connStat.IPTranslation.ReplSrcIP
+				key.Port = connStat.IPTranslation.ReplSrcPort
+			}
+		} else {
+			key.Address = connStat.Source
```

Comment:
We don't need to take the IPTranslation in to account for the source IP?

Reply:
No, because of how NAT connections show up for intra-host connections, the original IP is the "real" IP and the NATed source is the "virtual" IP.